File: Rich_Game/src/com/tw/giftHouse/GiftHouse.java
package com.tw.giftHouse;

import com.tw.asest.AssistancePower;
import com.tw.house.House;
import com.tw.player.Player;

/**
 * Created by pzzheng on 11/13/16.
 */
public class GiftHouse extends House {

    public GiftHouse(AssistancePower... gifts) {
        super(gifts);
    }

    @Override
    public Player.Status comeHere(Player player) {
        player.moveTo(this);
        return player.waitForResponse();
    }
}

File: ruby/rich_game_ruby/src/places/gift_house.rb
require_relative 'place'
require_relative '../commands/command_factory'
class GiftHouse < Place
  def initialize(*gift)
    @gifts = gift
  end

  def comeHere(player)
    player.moveTo(self)
    # return player.endTurn if !canAffordWith(player.asset.points)
    player.lastResponsiveCommand = CommandFactory::SelectGift
    return player.waitForResponse
  end

end